build.gradle

```diff
@@ -113,6 +114,19 @@ sonarqube {
   }
 }
 
+pmd {
+  toolVersion = "5.8.1"
+  ignoreFailures = true
+  sourceSets = [sourceSets.main]
```

Comment:
add `test` and `functionalTest`

Reply:
`pmd` for all sources is run during `check` task, and `pmdFunctionalTests` is also available.

Reply:
This property is used to configure what source sets should be analyzed as part of the `check` and `build` tasks.